http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ZombieRuntimeMonitor.scala

```diff
@@ -48,37 +54,70 @@ class ZombieRuntimeMonitor[F[_]: Parallel: ContextShift: Timer](
       semaphore <- Semaphore[F](config.concurrency)
 
-      // Get active clusters from the Leo DB, grouped by project
-      clusterMap <- ZombieMonitorQueries.listZombieQuery.transaction
+      // Get active runtimes from the Leo DB, grouped by project
+      activeRuntimeMap <- ZombieMonitorQueries.listActiveZombieQuery.transaction
 
       _ <- logger.info(
-        s"Starting zombie detection across ${clusterMap.size} projects with concurrency of ${config.concurrency}"
+        s"Starting active zombie detection across ${activeRuntimeMap.size} projects with concurrency of ${config.concurrency}"
       )
-      zombies <- clusterMap.toList.parFlatTraverse[F, PotentialZombieRuntime] {
-        case (project, clusters) =>
+      activeZombies <- activeRuntimeMap.toList.parFlatTraverse[F, ZombieCandidate] {
+        case (project, zombieCandidates) =>
           semaphore.withPermit(
             // Check if the project is active
             isProjectActiveInGoogle(project).flatMap {
               case true =>
-                // If the project is active, check each individual cluster
-                clusters.toList.traverseFilter { cluster =>
-                  isRuntimeActiveInGoogle(cluster, startInstant).ifA(F.pure(None), F.pure(Some(cluster)))
+                // If the project is active, check each individual runtime
+                zombieCandidates.toList.traverseFilter { candidate =>
+                  // If the runtime is less than one minute old, it may not exist in Google because it is still Provisioning, so it's not a zombie
+                  isRuntimeActiveInGoogle(candidate.googleProject, candidate.runtimeName, candidate.cloudService, startInstant)
+                    .ifA[Option[ZombieCandidate]](F.pure(None), F.pure(zombieIfOlderThanOneMinute(candidate, startInstant)))
                 }
               case false =>
-                // If the project is inactive, all clusters in the project are zombies
-                F.pure(clusters.toList)
+                // If the project is inactive, all runtimes in the project are zombies
+                F.pure(zombieCandidates.toList)
             }
           )
       }
-      // Error out each detected zombie cluster
-      _ <- zombies.parTraverse(zombie => semaphore.withPermit(handleZombieRuntime(zombie, startInstant)))
+
+      // Get all deleted runtimes that haven't been confirmed from the Leo DB
+      unconfirmedDeletedRuntimes <- ZombieMonitorQueries.listInactiveZombieQuery.transaction
+
+      _ <- logger.info(
+        s"Starting inactive zombie detection within ${unconfirmedDeletedRuntimes.size} runtimes with concurrency of ${config.concurrency}"
+      )
+
+      inactiveZombies <- unconfirmedDeletedRuntimes.parTraverse {
+        candidate =>
+          semaphore.withPermit {
+            isRuntimeActiveInGoogle(candidate.googleProject, candidate.runtimeName, candidate.cloudService, startInstant).flatMap { isActive =>
```

Comment:
Is there a more scala-esque way to do this? `ifA` didn't work here since it looks like it evaluates both parameters and returns the relevant one, which meant that the database was being updated regardless of the boolean value

Reply:
I don't think so...you're probably have to use `flatMap` and then if else